Controller/Payment/Checkout/Token.php

```diff
@@ -122,4 +141,28 @@ public function execute()
             ->setHttpResponseCode(\Magento\Framework\Webapi\Response::HTTP_OK)
             ->setData($resBody);
     }
+
+    /**
+     * CreateBuyer
+     *
+     * @param $transactionId
+     */
+    public function createBuyer($transactionId)
+    {
+        try {
+            if ($this->customerSession->isLoggedIn() && empty($this->helper->getBuyerId()) && $transactionId) {
```

Comment:
move condition outside of this method – to keep method only responsible for creating the buyer